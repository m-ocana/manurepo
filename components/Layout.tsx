import Head from "next/head";
import Link from "next/link";
import Header from "./Header";
import BackToHome from "./BackToHome";
import utilStyles from "../styles/utils.module.css";
import styled from "styled-components";
import { TITLE } from "../globals/site";

const Container = styled.div`
  max-width: 36rem;
  padding: 0 1rem;
  margin: 3rem auto 6rem;
`;

export default function Layout({
  children,
  home,
}: {
  children: React.ReactNode;
  home?: boolean;
}) {
  return (
    <Container>
      <Head>
        <link rel="icon" href="/favicon.ico" />
        <meta
          name="description"
          content="Collection of cheatsheets, snippets and utils"
        />
        <meta name="og:title" content={TITLE} />
      </Head>
      <Header home={home} />
      <main>{children}</main>
      {!home && <BackToHome />}
    </Container>
  );
}
