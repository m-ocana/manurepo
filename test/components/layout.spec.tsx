import React from "react";
import { render, act } from "@testing-library/react";
import Layout from "../../components/layout";

describe("Layout", () => {
  it("scaffolds the home page", () => {
    const rendered = render(<Layout home children={null} />);
    expect(rendered.asFragment()).toMatchSnapshot();
  });

  it("scaffolds any other page", () => {
    const rendered = render(<Layout children={null} />);
    expect(rendered.asFragment()).toMatchSnapshot();
  });
});
