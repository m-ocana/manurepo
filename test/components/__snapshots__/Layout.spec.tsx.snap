// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout scaffolds any other page 1`] = `
<DocumentFragment>
  <div
    class="Layout__Container-sc-1iaoog8-0 kefHsW"
  >
    <header
      class="Header__HeaderContainer-sc-1h5jhiv-0 fPyefU"
    >
      <a
        class="Header__LinkWrapper-sc-1h5jhiv-3 dXWzUE"
      >
        <img
          alt="The Manurepo"
          class="Header__HeaderImage-sc-1h5jhiv-1 dppRBg"
          src="/images/repo.svg"
        />
        <h2
          class="Header__H2-sc-1h5jhiv-4 dgzwDa"
        >
          The Manurepo
        </h2>
      </a>
    </header>
    <main />
    <div
      class="BackToHome__BackToHomeContainer-sc-1b2or72-0 bzyFly"
    >
      <a
        href="/"
      >
        ‚Üê Back to home
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Layout scaffolds the home page 1`] = `
<DocumentFragment>
  <div
    class="Layout__Container-sc-1iaoog8-0 kefHsW"
  >
    <header
      class="Header__HeaderContainer-sc-1h5jhiv-0 fPyefU"
    >
      <img
        alt="The Manurepo"
        class="Header__HeaderImage-sc-1h5jhiv-1 bASMFq"
        src="/images/repo.svg"
      />
      <h1
        class="Header__H1-sc-1h5jhiv-2 divOda"
      >
        The Manurepo
      </h1>
    </header>
    <main />
  </div>
</DocumentFragment>
`;
